src/picongpu/include/particles/ionization/byField/ADK/AlgorithmADK.hpp

```diff
@@ -0,0 +1,103 @@
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "types.h"
+#include "simulation_defines.hpp"
+#include "particles/traits/GetAtomicNumbers.hpp"
+#include "traits/attribute/GetChargeState.hpp"
+#include "algorithms/math/floatMath/floatingPoint.tpp"
+#include "plugins/radiation/utilities.hpp"
+
+/** \file AlgorithmADK.hpp
+ *
+ * IONIZATION ALGORITHM for the ADK model
+ *
+ * - implements the calculation of ionization probability and changes charge states
+ *   by decreasing the number of bound electrons
+ * - is called with the IONIZATION MODEL, specifically by setting the flag in @see speciesDefinition.param */
+
+namespace picongpu
+{
+namespace particles
+{
+namespace ionization
+{
+
+    /** \struct AlgorithmADK
+     *
+     * \brief calculation for the Ammosov-Delone-Krainov tunneling model */
+    struct AlgorithmADK
+    {
+
+        /** Functor implementation
+         * \tparam EType type of electric field
+         * \tparam BType type of magnetic field
+         * \tparam ParticleType type of particle to be ionized
+         *
+         * \param bField magnetic field value at t=0
+         * \param eField electric field value at t=0
+         * \param parentIon particle instance to be ionized with position at t=0 and momentum at t=-1/2
+         */
+        template<typename EType, typename BType, typename ParticleType >
+        HDINLINE void
+        operator()( const BType bField, const EType eField, ParticleType& parentIon, float_X randNr )
+        {
+
+            const float_X protonNumber  = GetAtomicNumbers<ParticleType>::type::numberOfProtons;
+            float_X chargeState         = attribute::getChargeState(parentIon);
+
+            uint32_t cs         = math::float2int_rd(chargeState);
+            const float_X pi    = precisionCast<float_X>(M_PI);
+            /* electric field in atomic units - only absolute value */
+            float_X eInAU       = math::abs(eField)*UNIT_EFIELD / SI::ATOMIC_UNIT_EFIELD;
+
+            /* effective principal quantum number */
+            float_X nEff        = protonNumber / math::sqrt(2*SI::IONIZATION_ENERGY[cs]/SI::ATOMIC_UNIT_ENERGY);
+            /* nameless variable for convenience dFromADK*/
+            float_X dBase       = 4 * util::cube(protonNumber) / (( eInAU ) * util::quad(nEff)) ;
+            float_X dFromADK    = math::pow(dBase,nEff);
+
+            /* ionization rate */
+            float_X rateADK     = math::sqrt(3 * util::cube(nEff) * eInAU / (pi * util::cube(protonNumber))) \
+                                    * eInAU * util::square(dFromADK) / (8 * pi * protonNumber) \
+                                    * math::exp(-2 * util::cube(protonNumber) / (3 * util::cube(nEff) * eInAU));
+
+            /* ionization probability
+             * calculated like a Poissonian induced process
+             */
+            float_X probADK     = rateADK * SI::DELTA_T_SI/SI::ATOMIC_UNIT_TIME;
+            if (probADK > 0.1)
```

Comment:
so should I rather do 
```
if ( probADK > float_X(0.1))
```
or even save that value in an extra variable?

I am doing this btw to save some compute time because I am calculating only the first term of the Taylor series expansion for the ionization probability as long as the error doesn't surpass 1%.